### PR TITLE
Implement nats avro serdes

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

File: serdes/nats/avro-serde/src/main/java/io/apicurio/registry/serde/avro/nats/client/streaming/producers/NatsProducer.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.apicurio.registry.serde.avro.nats.client.streaming.producers;
+
+import io.apicurio.registry.serde.avro.nats.client.exceptions.ApicurioNatsException;
+
+public interface NatsProducer<T> extends AutoCloseable {
+
+    void send(T message) throws ApicurioNatsException;
```
Comment (sourabhaggrawal) [inline serdes/nats/avro-serde/src/main/java/io/apicurio/registry/serde/avro/nats/client/streaming/producers/NatsProducer.java:7]: let's rename it to publish and add publishAsync too ?

**Comment by carlesarnal** [comment]
Good comments, I'm going to create an issue so we can track them better.

**Comment by carlesarnal** [comment]
@sourabhaggrawal I have created the following GH issue to track your comments: https://github.com/Apicurio/apicurio-registry/issues/5215. Would you be open to contribute a PR to address them?

**Comment by souravagrawal** [comment]
Thank you @carlesarnal,  I have submitted PR for changes https://github.com/Apicurio/apicurio-registry/pull/5222

**Pull request merged**